codegen/projections/rails_json/lib/rails_json/builders.rb

```diff
@@ -515,15 +515,21 @@ def self.build(http_req, input:)
         http_req.headers['X-Double'] = Hearth::NumberHelper.serialize(input[:header_double]) unless input[:header_double].nil?
         http_req.headers['X-Boolean1'] = input[:header_true_bool].to_s unless input[:header_true_bool].nil?
         http_req.headers['X-Boolean2'] = input[:header_false_bool].to_s unless input[:header_false_bool].nil?
-        http_req.headers['X-StringList'] = input[:header_string_list] unless input[:header_string_list].nil? || input[:header_string_list].empty?
-        http_req.headers['X-StringSet'] = input[:header_string_set] unless input[:header_string_set].nil? || input[:header_string_set].empty?
-        http_req.headers['X-IntegerList'] = input[:header_integer_list] unless input[:header_integer_list].nil? || input[:header_integer_list].empty?
-        http_req.headers['X-BooleanList'] = input[:header_boolean_list] unless input[:header_boolean_list].nil? || input[:header_boolean_list].empty?
+        unless input[:header_string_list].nil? || input[:header_string_list].empty?
+          http_req.headers['X-StringList'] = Hearth::Http::HeaderListBuilder.build_string_list(input[:header_string_list])
+        end
+        unless input[:header_string_set].nil? || input[:header_string_set].empty?
+          http_req.headers['X-StringSet'] = Hearth::Http::HeaderListBuilder.build_string_list(input[:header_string_set])
+        end
+        http_req.headers['X-IntegerList'] = input[:header_integer_list].compact.join(', ') unless input[:header_integer_list].nil? || input[:header_integer_list].empty?
+        http_req.headers['X-BooleanList'] = input[:header_boolean_list].compact.join(', ') unless input[:header_boolean_list].nil? || input[:header_boolean_list].empty?
         unless input[:header_timestamp_list].nil? || input[:header_timestamp_list].empty?
-          http_req.headers['X-TimestampList'] = input[:header_timestamp_list].map { |t| Hearth::TimeHelper.to_http_date(t) }.join(', ')
+          http_req.headers['X-TimestampList'] = input[:header_timestamp_list].compact.map { |t| Hearth::TimeHelper.to_http_date(t) }.join(', ')
```

Comment:
I think the method for this in the header list builder would be "build_http_date_list"